### PR TITLE
(Chore) Include final form in step two

### PR DESCRIPTION
**Pull request by fernandomg**
This PR is the first step in order to complete #711

Final Form is a library that will help us improve form element's validation and manipulation across the whole wizard.

[`react-final-form`](https://github.com/final-form/react-final-form) in particular is a react component with it's own `state` that facilitates the interaction with the elements of a form. Makes easier and cleaner the event and error handling, field validations, and all things related to forms in general.

Allows cross evaluations, computed values, conditional fields, auto-focus on the first errored field, etc.

---

In this PR I worked on the `StepTwo`, and migrated `name`, `ticker` and `decimals` to use this library.

As a result StepTwo ended up being more concise and clear and form elements became easier to test.

I think that we should switch a bit the focus of #711 and point it to improve Token Wizard form handling.

@vbaranov please let me know what you think about it.

### CODE REVIEW

File: src/components/stepTwo/index.js [hunk 1]
```diff
@@ -1,142 +1,78 @@
...
-    }
+  updateTokenStore = (property, value) => {
+    const validity = validators(property, value) ? VALID : INVALID
 
-    this.props.reservedTokenStore.addToken(newToken)
+    this.props.tokenStore.setProperty(property, value)
+    this.props.tokenStore.updateValidity(property, validity)
+  }
+
+  onSubmit = () => {
```
Comment (fvictorio) [inline src/components/stepTwo/index.js]: This isn't doing anything right now, is it?

File: src/components/stepTwo/StepTwoForm.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+const errorStyle = {
+  color: 'red',
+  fontWeight: 'bold',
+  fontSize: '12px',
+  width: '100%',
+  height: '10px',
+}
+
+@observer
```
Comment (fvictorio) [inline src/components/stepTwo/StepTwoForm.js]: Does this need to be an observer?

File: src/components/Common/TokenDecimals.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { validateDecimals } from '../../utils/validations'
+import { TEXT_FIELDS } from '../../utils/constants'
+import { Field } from 'react-final-form'
+import { InputField2 } from './InputField2'
+
+export const TokenDecimals = ({ disabled, errorStyle }) => (
+  <Field
+    validate={validateDecimals}
+    component={InputField2}
+    parse={(value) => /^[0-9]*$/.test(value) ? value : ''}
```
Comment (fvictorio) [inline src/components/Common/TokenDecimals.js]: This causes the whole input to be cleared if some typed character is invalid.

File: src/components/stepTwo/index.js [hunk 4]
```diff
@@ -1,142 +1,78 @@
...
-  }
-
-  renderLinkComponent = () => {
-    if(this.props.tokenStore.isTokenValid) {
-      return this.renderLink();
-    }
-    return <div onClick={this.showErrorMessages.bind(this, 'token')} className="button button_fill"> Continue</div>
+    document
+      .getElementById('tokenData')
+      .dispatchEvent(new Event('submit'))
```
[the remark this answers is not in the document]
Comment (fernandomg) [reply src/components/stepTwo/index.js]: indeed

File: src/components/stepTwo/index.js (hunk 1 above)
Comment (fernandomg) [reply src/components/stepTwo/index.js]: `onSubmit`? when the `submit` event is [dispatched](https://github.com/poanetwork/token-wizard/pull/719/files/a4970602f816d6a477070fcc160b02c0c8a8bf73#diff-c0618d3a6e868cb93320cafc2d83a613R42), if all the the fields in the form are valid, this method [will be called](https://github.com/poanetwork/token-wizard/pull/719/files/a4970602f816d6a477070fcc160b02c0c8a8bf73#diff-c0618d3a6e868cb93320cafc2d83a613R90)

File: src/components/stepTwo/StepTwoForm.js (hunk 2 above)
Comment (fernandomg) [reply src/components/stepTwo/StepTwoForm.js]: It wasn't necessary. Converted it to a functional component.

File: src/components/Common/TokenDecimals.js (hunk 3 above)
Comment (fernandomg) [reply src/components/Common/TokenDecimals.js]: Yes, just refactored. Now works as expected.

**Comment by fvictorio** [comment]
@vbaranov did you have a chance to test this? If we go with this library (and I think we should), almost all issues in our to-do list are going to depend on this PR, so it would be good to have it merged today.

**Comment by vbaranov** [comment]
@fvictorio not yet, I'll do it soon. Thanks for clarifying.

**Pull request merged**